modules/foreman_domain.py

```diff
@@ -121,35 +120,23 @@
 
 
 def main():
-    module = AnsibleModule(
+    module = ForemanAnsibleModule(
         argument_spec=dict(
-            server_url=dict(required=True),
-            username=dict(required=True),
-            password=dict(required=True, no_log=True),
-            verify_ssl=dict(type='bool', default=True),
             name=dict(required=True),
             description=dict(),
             dns_proxy=dict(),
             locations=dict(type='list'),
             organizations=dict(type='list'),
-            state=dict(choices=['present', 'absent'], default='present'),
         ),
         supports_check_mode=True,
     )
 
-    if has_import_error:
-        module.fail_json(msg=import_error_msg)
-
-    domain_dict = dict(
-        [(k, v) for (k, v) in module.params.items() if v is not None])
+    (server_params, domain_dict) = module.parse_params()
 
-    server_url = domain_dict.pop('server_url')
-    username = domain_dict.pop('username')
-    password = domain_dict.pop('password')
-    verify_ssl = domain_dict.pop('verify_ssl')
     state = domain_dict.pop('state')
```

Comment:
Should state also be popped off in `parse_params` and returned in the tuple?

Reply:
yes